src/analyzer/analyzerbeats.cpp

```diff
@@ -38,7 +39,7 @@ bool AnalyzerBeats::initialize(TrackPointer tio, int sampleRate, int totalSample
         return false;
     }
 
-    bool bPreferencesBeatDetectionEnabled = static_cast<bool>(
+    bool bPreferencesBeatDetectionEnabled = m_batch || static_cast<bool>(
```

Comment:
I don't know either why such option existed. I simply changed it to a constructor parameter rather than changing it on analysisfeature. I am open to talk about the need or meaning of it.